asserts/database.go

```diff
@@ -22,12 +22,34 @@
 package asserts
 
 import (
-	"errors"
 	"fmt"
 	"regexp"
 	"time"
 )
 
+// NotFoundError is returned when an assertion can not be found.
+type NotFoundError struct {
+	Type       *AssertionType
+	PrimaryKey []string
```

Comment:
Hmm, I'm not sure it's necessarily the *primary key* that was not found. It's potentially any set of headers, including the empty set IIRC.